modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -1945,29 +1365,84 @@ public Set<NodeWithAttributes> logicalTopology() {
         return logicalTopology;
     }
 
-    /**
-     * Returns the distribution zone name from the configuration, {@code null} if the zone is absent.
-     *
-     * @param configZoneId Distribution zone id from configuration.
-     */
-    // TODO: IGNITE-20114 Get rid of
-    @Deprecated(forRemoval = true)
-    public @Nullable String getZoneName(int configZoneId) {
-        DistributionZonesView zonesView = zonesConfiguration.value();
-
-        return Stream.concat(Stream.of(zonesView.defaultDistributionZone()), zonesView.distributionZones().stream())
-                .filter(zoneView -> zoneView.zoneId() == configZoneId)
-                .map(DistributionZoneView::name)
-                .findFirst()
-                .orElse(null);
+    private void registerCatalogEventListenersOnStartManagerBusy() {
+        catalogManager.listen(ZONE_CREATE, (parameters, exception) -> inBusyLock(busyLock, () -> {
+            assert exception == null : parameters;
+
+            CreateZoneEventParameters params = (CreateZoneEventParameters) parameters;
+
+            createOrRestoreZoneStateBusy(params.zoneDescriptor(), params.causalityToken());
+
+            return completedFuture(false);
+        }));
+
+        catalogManager.listen(ZONE_DROP, (parameters, exception) -> inBusyLock(busyLock, () -> {
+            assert exception == null : parameters;
+
+            onDropZoneBusy((DropZoneEventParameters) parameters);
+
+            return completedFuture(false);
+        }));
+
+        catalogManager.listen(ZONE_ALTER, new ManagerCatalogAlterZoneEventListener());
+    }
+
+    private void startZonesOnStartManagerBusy() {
+        CompletableFuture<Long> recoveryFinishedFuture = metaStorageManager.recoveryFinishedFuture();
+
+        assert recoveryFinishedFuture.isDone();
+
+        int catalogVersion = catalogManager.latestCatalogVersion();
+        long causalityToken = recoveryFinishedFuture.join();
+
+        for (CatalogZoneDescriptor zone : catalogManager.zones(catalogVersion)) {
```

Comment:
We should also cleanup all abandoned resources. Both local (vault) and distributed (ms based keys). I would rather create two separate tickets for both that items in order not to clutter up given PR, however they should be ranked as important ones.

Reply:
add todo